docroot/modules/custom/va_gov_media/src/Service/PdfDeleteService.php

```diff
@@ -0,0 +1,245 @@
+    // Used for name csv and directory.
+    $timestamp = date('Y-m-d-H:i:s');
+
+    foreach ($pdf_list as $pdf_id) {
+      $is_used = $this->usage->getUsageTotal('media', $pdf_id);
+      /** @var \Drupal\media\MediaInterface $pdf */
+      $pdf = $media_storage->load($pdf_id);
+      $pdf_name = $pdf->get('name')->value;
+      if (!$pdf->hasField('field_document')) {
+        $logger->warning("PDF {$pdf_id} not deleted because it has no document field.");
+        continue;
+      }
+      if ($is_used === 0 && $pdf->hasField('field_document')) {
+        $section_id = $pdf->get('field_owner')->getValue()[0]['target_id'];
+        if (!$section_id) {
+          $logger->warning("PDF {$pdf_id} has no section id.");
+          continue;
+        }
+        $section = $this->entityTypeManager->getStorage('taxonomy_term')->load($section_id);
+        $parents = $this->entityTypeManager->getStorage('taxonomy_term')->loadAllParents($section_id);
+        $parents = array_keys($parents);
+        $section_name = $section ? $section->get('name')->value : 'No Section';
+        // 3 is the Term ID for VHA.
+        if (in_array('8', $parents)) {
+          $user_id = $pdf->get('uid')->getValue()[0]['target_id'];
+          $user = $this->entityTypeManager->getStorage('user')->load($user_id);
+          $username = $user ? $user->getAccountName() : 'Unknown';
+          $created_time = $this->dateFormatter->format($pdf->get('created')->value, 'short');
+          $deleted_pdfs[] = [
+            $pdf_name,
+            $section_name,
+            $created_time,
+            $username,
+          ];
+          $this->deletePdf($pdf, $timestamp);
+          $vha_total++;
+
+        }
+        else {
+          $section_error = $section ? $section->get('name')->value : $section_id;
+          $logger->info("PDF ({$pdf_name}) not deleted because it is not in a VHA section (Name or ID: {$section_error}).");
+          $non_vha_total++;
+        }
+      }
+    }
+    $this->writeCsv($deleted_pdfs, $timestamp);
+    $logger->info('PDFs deleted and CSV written.');
+    $logger->info("Total PDFs: {$total_pdfs}");
+    $logger->info("VHA unattached PDFs deleted: {$vha_total}");
+    $logger->info("Non-VHA unattached PDFs not deleted: {$non_vha_total}");
+  }
+
+  /**
+   * Delete PDF.
+   *
+   * @param \Drupal\media\MediaInterface $pdf_entity
+   *   The PDF entity to delete.
+   * @param string $timestamp
+   *   The timestamp to use for the directory.
+   *
+   * @return void
+   *   Return void.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function deletePdf(MediaInterface $pdf_entity, string $timestamp): void {
+    $file = $pdf_entity->field_document->entity;
+    if (!$file instanceof FileInterface) {
+      $pdf_name = $pdf_entity->get('name')->value;
+      $this->loggerFactory->get('va_gov_media')->warning("PDF {$pdf_name} file could not be copied because it has no file entity.");
```

Comment:
If I understand this correctly, we are deleting the PDF _entity_ here because it has no PDF file attached to it. If that's the case, I would amend the logged message as below.

```suggestion
      $this->loggerFactory->get('va_gov_media')->warning("{$pdf_name} does not have a file attached to it. No file was copied. Deleting the file entity.");
```